src/test/java/com/microsoft/sqlserver/jdbc/requestboundary/RequestBoundaryMethodsTest.java

```diff
@@ -0,0 +1,421 @@
+            if (null != stmt) {
+                Utils.dropTableIfExists(tableName, stmt);
+                stmt.close();
+            }
+            if (null != con) {
+                con.close();
+            }
+        }
+    }
+
+    /**
+     * Tests Request Boundary methods with statements.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testStatements() throws SQLException {
+        Statement stmt1 = null;
+        PreparedStatement ps = null;
+        CallableStatement cs = null;
+        ResultSet rs1 = null;
+
+        try {
+            con = connect();
+
+            if (Utils.isJDBC43AndGreater(con)) {
+                stmt1 = con.createStatement();
+                con.beginRequest();
+                stmt = con.createStatement();
+                rs = stmt.executeQuery("SELECT 1");
+                rs.next();
+                assertEquals(1, rs.getInt(1));
+                con.endRequest();
+
+                assertTrue(!stmt1.isClosed(), "Statement created outside of beginRequest()/endRequest() block should not be closed.");
+                assertTrue(stmt.isClosed(), "Statment created inside beginRequest()/endRequest() block should be closed after endRequest().");
+                assertTrue(rs.isClosed(), "ResultSet should be closed after endRequest().");
+                stmt1.close();
+
+                // Multiple statements inside beginRequest()/endRequest() block
+                con.beginRequest();
+                stmt = con.createStatement();
+                String tableName = AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("RequestBoundary"));
+                Utils.dropTableIfExists(tableName, stmt);
+                stmt.executeUpdate("CREATE TABLE " + tableName + " (col int)");
+                ps = con.prepareStatement("INSERT INTO " + tableName + " values (?)");
+                ps.setInt(1, 2);
+                ps.executeUpdate();
+
+                stmt1 = con.createStatement();
+                rs1 = stmt1.executeQuery("SELECT * FROM " + tableName);
+                rs1.next();
+                assertEquals(2, rs1.getInt(1));
+                Utils.dropTableIfExists(tableName, stmt);
+
+                cs = con.prepareCall("{call sp_server_info}");
+                cs.execute();
+                con.endRequest();
+
+                assertTrue(stmt.isClosed());
+                assertTrue(ps.isClosed());
+                assertTrue(stmt1.isClosed());
+                assertTrue(cs.isClosed());
+                assertTrue(rs1.isClosed());
+            }
+        }
+        finally {
+            if (null != stmt) {
+                stmt.close();
+            }
+            if (null != stmt1) {
+                stmt1.close();
+            }
+            if (null != ps) {
+                ps.close();
+            }
+            if (null != cs) {
+                cs.close();
+            }
+            if (null != con) {
+                con.close();
+            }

```

Comment:
Please use try-with-resources in all tests.

Reply:
The method tests whether statements are closed after `endRequest()`. This should be fine.

Reply:
I think if you're testing statments closed or not, you can use _asserts_ after endRequest is called. 
`Assert.assertTrue(closeable.isClosed());`

And connecion is not part of test, so it should be always in try-with-resources block. Basically, any closeable object created in tests must always be created in try-with-resources blocks to avoid resource leaks if tests fail due to any reason.

Reply:
Asserts are already there.

I agree that we should use try-with-resources when possible, but for this scenario I will keep the finally block instead, because the method tests multiple statements between `beginRequest()`/ `endRequest()` and we would need to add multiple try blocks too. Which in my opinion will make the code less readable.